packages/core/src/config.ts

```diff
@@ -1434,6 +1435,17 @@ const composeBundlelessExternalConfig = (
                     }
                   }
                 } else {
+                  // If the import path refers to a directory,
+                  // it most likely actually refers to a `index.*` file due to Node's module resolution.
+                  // When redirect.js.path is set to false, index should still be added before adding extension.
+                  // When redirect.js.path is true, the resolver directly generate correct resolvedRequest with index appended.
+                  if (
+                    !jsRedirectPath &&
+                    (await isDirectory(join(dirname(issuer), resolvedRequest)))
```

Comment:
i wonder if this will be a bit perf harmful as we will invoke this function for almost every file?